plugins/processors/execd/execd.go

```diff
@@ -0,0 +1,147 @@
+package execd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal/process"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/serializers"
+)
+
+const sampleConfig = `
+  ## Program to run as daemon
+  command = ["telegraf-smartctl", "-d", "/dev/sda"]
+
+  ## Delay before the process is restarted after an unexpected termination
+  restart_delay = "10s"
+`
+
+type Execd struct {
+	Command      []string
+	RestartDelay config.Duration
+
+	parserConfig     *parsers.Config
+	parser           parsers.Parser
+	serializerConfig *serializers.Config
```

Comment:
How do you set the parser/serializer config?

Reply:
I have no idea. Open to suggestions here. Maybe *Config needs to be public with toml tags. At the moment I don't think anything other than influx is going to work for this processor, so it may be moot. json has the issue we previously discussed where the parse format doesn't match the serialize format. A new json parser/serializer pair would resolve that, if we're willing to go that direction.